src/Microsoft.Health.Fhir.Core.UnitTests/Models/PartialDateTimeTests.cs

```diff
@@ -14,172 +14,133 @@ namespace Microsoft.Health.Fhir.Core.UnitTests.Models
+            Assert.Equal(expectedMessage, ex.Message);
         }
 
-        public static IEnumerable<object[]> GetParameterOutOfRangeData()
+        [Theory]
+        [InlineData("2013-05-18T23:09+1:00")]
+        [InlineData("2013-05-18T23:09-4:30")]
+        [InlineData("2013-05-18T23:09+01:300")]
+        [InlineData("2013-05-18T23:09+01:77")]
+        [InlineData("2013-05-18T23:09+1")]
+        [InlineData("2013-05-18T23:09-7")]
+        [InlineData("2013-05-18T23:09-07")]
+        [InlineData("2013-05-18T23:09+Z")]
+        [InlineData("2013-05-18T23:09-Z")]
+        [InlineData("2013-05-18T23:09ZZ")]
+        [InlineData("2013-05-18T23:09-99:00")]
+        [InlineData("2013-05-18T23:09-789")]
+        public void GivenInvalidUtcOffset_WhenParsingPartialDateTime_ThenExceptionShouldBeThrown(string inputString)
         {
-            yield return new object[] { ParamNameYear, 0, 1, 1, 1, 1, 1, 0m, 60 }; // Year cannot be less than 1.
-            yield return new object[] { ParamNameYear, 10000, 1, 1, 1, 1, 1, 0m, 60 }; // Year cannot be greater than 9999.
-            yield return new object[] { ParamNameMonth, 2017, 0, 1, 1, 1, 1, 0m, 60 }; // Month cannot be less than 1.
-            yield return new object[] { ParamNameMonth, 2017, 13, 1, 1, 1, 1, 0m, 60 }; // Month cannot be greater than 12.
-            yield return new object[] { ParamNameDay, 2017, 1, 0, 1, 1, 1, 0m, 60 }; // Day cannot be less than 1.
-            yield return new object[] { ParamNameDay, 2017, 1, 32, 1, 1, 1, 0m, 60 }; // Day cannot be greater 31 in January.
-            yield return new object[] { ParamNameDay, 2001, 2, 29, 1, 1, 1, 0m, 60 }; // Day cannot be greater than 28 in non-leap year.
-            yield return new object[] { ParamNameDay, 2000, 2, 30, 1, 1, 1, 0m, 60 }; // Day cannot be greater than 29 in leap year.
-            yield return new object[] { ParamNameHour, 2017, 1, 1, -1, 1, 1, 0m, 60 }; // Hour cannot be less than 0.
-            yield return new object[] { ParamNameHour, 2017, 1, 1, 24, 1, 1, 0m, 60 }; // Hour cannot be greater than 23.
-            yield return new object[] { ParamNameMinute, 2017, 1, 1, 1, -1, 1, 0m, 60 }; // Minute cannot be less than 0.
-            yield return new object[] { ParamNameMinute, 2017, 1, 1, 1, 61, 1, 0m, 60 }; // Minute cannot be greater than 59.
-            yield return new object[] { ParamNameSecond, 2017, 1, 1, 1, 1, -1, 0m, 60 }; // Second cannot be less than 0.
-            yield return new object[] { ParamNameSecond, 2017, 1, 1, 1, 1, 61, 0m, 60 }; // Second cannot be greater than 59.
-            yield return new object[] { ParamNameFraction, 2017, 1, 1, 1, 1, 1, -1m, 60 }; // Fraction cannot be less than 0.
-            yield return new object[] { ParamNameFraction, 2017, 1, 1, 1, 1, 1, 1m, 60 }; // Fraction cannot be greater than .
+            Assert.Throws<FormatException>(() => PartialDateTime.Parse(inputString));
```

Comment:
Why doesn't this one check the message like the ones above do?

Reply:
Good point. I took it out because this exception can be thrown from two different places. They both send the same message, though, so I think it makes sense to add it in. Thanks!

Reply:
Fixed in 5818fc1.